File: 2/CS/MyText.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using MyString;

namespace MyText
{
    public class Mytext
    {
        int length = 0;
	private Mystring[] _text = new Mystring[0];

	//Adding MyString object in vector
	public void AddStr(char[] strToAdd)
	{
		Array.Resize(ref _text, _text.Length+1);
        Mystring word = new Mystring(strToAdd);
        _text[_text.Length - 1] = word;
	}

	//Deleting MyString object from vector
	public void DeleteStr(int strToDel)
	{
        _text = new Mystring[0];
	}

	//Clearing Text
	public void CleatText()
	{
		_text = new Mystring[0];
	}
	/*//Making first letters to upper register
	public void Toupper()
	{
		for (int i = 0; i < length; i++)
		{
            Mystring a = _text[i];
			a[0] = ToUpper(a[0]);
		}
	}

	//Find a key(the word that contain from first letters of the strings)
	public Mystring TakeAKey()
	{
		Mystring Key;
		for (int i = 0; i < length; i++)
		{
			Key[i] = _text[i][0];
			Key.Length++;
		}

		return Key;
	}
*/
	//Return amount of string with the number of chars
	public int TheLength(int amount)
	{
		int quantity = 0;
		for (int i = 0; i < length; i++)
		{
            int a = _text[i].MyLength();
			if ( a == amount)
			{
				quantity++;
			}
		}

		return quantity;
	}

	//Printing 'i' element of vector
	public void Print(int i)
	{
		_text[i].Print();
	}

	//Printing whole vector
	public void Print()
	{
		for (int i = 0; i < length; i++)
		{
			Print(i);
		}
	}

    }
}

File: 2/cpp/MyText.cpp
#include <iostream>
#include <string>
#include <vector>
#include "MyString.h"
#include "MyText.h"
using namespace std;



	int length = 0;
	vector<MyString> text;

	//Adding MyString object in vector
	void Text::AddStr(MyString strToAdd)
	{
		text.push_back(strToAdd);
		length++;
	}

	//Deleting MyString object from vector
	void Text::DeleteStr(int strToDel)
	{
		text.erase(text.begin() + strToDel - 1);
		length--;
	}

	//Clearing Text
	void Text::CleatText()
	{
		text.erase(text.begin(), text.end());
		length = 0;
	}

	//Making first letters to upper register
	void Text::ToUpper()
	{
		for (int i = 0; i < length; i++)
		{
			text[i][0] = toupper(text[i][0]);
		}
	}

	//Find a key(the word that contain from first letters of the strings)
	MyString Text::TakeAKey()
	{
		MyString Key;
		for (int i = 0; i < length; i++)
		{
			Key[i] = text[i][0];
			Key.length++;
		}

		return Key;
	}

	//Return amount of string with the number of chars
	int Text::TheLength(int amount)
	{
		int quantity = 0;
		for (int i = 0; i < length; i++)
		{
			if (text[i].length == amount)
			{
				quantity++;
			}
		}

		return quantity;
	}

	//For calling operator "[]"
	MyString& Text::operator [](int index)
	{
		return text[index];
	}

	//Printing 'i' element of vector
	void Text::Print(int i)
	{
		text[i].Print();
	}

	//Printing whole vector
	void Text::Print()
	{
		for (int i = 0; i < length; i++)
		{
			Print(i);
		}
	}
